doe/design.py

```diff
@@ -108,7 +108,7 @@ def find_local_max_ipopt(
         ):
             warnings.warn(
                 "Nonlinear constraints were detected. Not all features and checks are supported for this type of constraints. \
```

Comment:
Maybe something shorter?
"Nonlinear constraints were detected. Not all features and checks are supported for them. For better performance please make sure to provide their jacobians."